File: classic puzzle-medium/gravity-tumbler.py
#https://www.codingame.com/training/medium/gravity-tumbler
#GRAVITY TUMBLER
import re
import numpy as np
w,h=map(int,input().split())
count=int(input())
m=[]
for i in range(h):
    r=''.join(re.findall(r"#+",input()))
    m+=[list(r+(w-len(r))*".")]
#Use numpy to rotate the 2D matrix
arr=np.array(m)
for i in range(count):
    arr=np.rot90(arr)
    if i!=0: arr=arr[::-1]
for j in range(len(arr)):print(''.join(arr[j]))

File: classic puzzle-medium&above/gravity-tumbler.rb
#https://www.codingame.com/training/medium/gravity-tumbler
#GRAVITY TUMBLER
w,h=gets.split.map(&:to_i)
count=gets.to_i
#initialize the matrix
m=$<.map{
  s=_1.scan(/#+/).join
  s=s+"."*(w-s.size)
  s.chars
}
#rotate the matrix
for i in (0...count)
  t=m.transpose.map{_1.join}
  i==0?m=t.reverse.map{_1.chars}:m=t.map{_1.chars}
end
m.each{puts _1.join}
